instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-testing/src/main/groovy/JaxRsHttpServerTest.groovy

```diff
@@ -58,6 +59,36 @@ abstract class JaxRsHttpServerTest<S> extends HttpServerTest<S> {
     "canceled"   | "cancel"  | 503        | { it instanceof String } | true        | false   | null
   }
 
+  @Unroll
+  def "should handle #desc CompletionStage (JAX-RS 2.1+ only)"() {
+    assumeTrue(supportsJaxRs21())
+
+    given:
+    def url = HttpUrl.get(address.resolve("/async-completion-stage")).newBuilder()
+      .addQueryParameter("action", action)
+      .build()
+    def request = request(url, "GET", null).build()
+
+    when:
+    def response = client.newCall(request).execute()
```

Comment:
IIUC, it's important our test makes sure a span hasn't been ended before the future is completed. How about adding some sleep in the API method, and separate out making the call, checking there are no traces (with previous code I think there would have been), than joining on the future, then checking there are traces?

Reply:
Instead of adding sleep (which would make this test at least a bit flaky) I've added synchronisation using a `CyclicBarrier` -- but the principle is the same: no traces until the future is complete.